tests/integration.rs

```diff
@@ -983,7 +986,7 @@ trigger = { type = "http", base = "/" }
 version = "0.1.0"
 [[component]]
 id = "unbuilt"
-source = "DOES-NOT-EXIST.wasm"
+source = "spin.toml" # dirty trick
```

Comment:
This appears to be changing the test - the test is meant to assert that help works even when the app has not been build (i.e. `source` failed to resolve _at all_).  I.e. that `spin new > spin up --help` works correctly.  This seems (if I'm reading it right) to be using a file that exists but isn't valid Wasm, which isn't what this test wants to prove   What am I missing?

Reply:
Ah, I didn't realize this was testing that intentionally; indeed this is a regression for that situation. The basic problem is that deep in `spin_trigger::locked::build_locked_app` we call `canonicalize` on all our paths, which requires those paths to exist. Maybe I can resolve that problem...

Reply:
Yes, I think that was something Brian hit when trying to unify around the locked app, and is (at least part of) why the local-vs-remote code paths came to diverge.

Reply:
Certainly the local loader tries to absolutise paths as early as possible.

One of these days I want `AbsolutePath` and `RelativePath` types so we can document this behaviour at the type level _and actually enforce_ that it remains true...

Reply:
I was just testing this with a malformed `spin.toml` that is an `http` trigger, not sure if this  has been resolved just yet but i see the following notably missing the trigger options:

```❯  spin up -f examples/http-rust -h
spin-up
Start the Spin application

USAGE:
    spin up [OPTIONS]

OPTIONS:
        --direct-mounts         For local apps with directory mounts and no excluded files, mount
                                them directly instead of using a temporary directory
    -e, --env <ENV>             Pass an environment variable (key=value) to all components of the
                                application
    -f, --from <APPLICATION>    The application to run. This may be a manifest (spin.toml) file, a
                                directory containing a spin.toml file, or a remote registry
                                reference. If omitted, it defaults to "spin.toml"
    -h, --help
    -k, --insecure              Ignore server certificate errors from bindle server or registry
        --temp <TMP>            Temporary directory for the static assets of the components
```

Reply:
Ugh, good catch.  I bet an actually malformed `spin.toml` is a whole new failure mode.  Guessing it gets past the "no app so show default trigger options" stage, but then can't deserialise the file to work out the trigger and so gives up rather than calling to help-only trigger.  It's an existing problem though, not an issue with this PR; I'd do it as a separate issue.